openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -225,3 +156,275 @@ jint tcn_get_java_env(JNIEnv **env)
     }
     return JNI_OK;
 }
+
+// TODO: Share code with netty natives utilities.
+char* netty_internal_tcnative_util_prepend(const char* prefix, const char* str) {
+    if (prefix == NULL) {
```

Comment:
I would prefer not to use format based based methods when non-format methods can be easily used instead.